contracts/MANACrowdsale.sol

```diff
@@ -41,16 +42,22 @@ contract MANACrowdsale is ContinuousCrowdsale, WhitelistedCrowdsale, CappedCrowd
     {
         require(_rateStepDecrease > 0);
         require(_preferentialRate > 0);
-        require(_rate > _rateStepDecrease * (_endBlock - _startBlock));
+        require(_rate > _rateStepDecrease * (1 + _endBlock - _startBlock));
 
         rateStepDecrease = _rateStepDecrease;
         preferentialRate = _preferentialRate;
+
+        continuousSale = createContinuousSaleContract();
     }
 
-    function createTokenContract() internal returns (MintableToken) {
+    function createTokenContract() internal returns(MintableToken) {
         return new MANAToken();
     }
 
+    function createContinuousSaleContract() internal returns(MANAContinuousSale) {
+        return new MANAContinuousSale(rate, wallet, token);
+    }
+
     function setBuyerRate(address buyer, uint256 rate) onlyOwner public {
         require(buyer != 0);
         require(rate != 0);
```

Comment:
This is lacking two sanity checks:
- require buyer to be whitelisted
- require crowdsale to have not started yet

It would also be good to emit an event for transparency